<?xml version="1.0"?>
<launch>
    <!--Save images and calibration data too folder - default "" = ~/calibration_images/M_D/N-->
    <arg name="save_directory" default=""/>

    <!--Image topic names-->
    <arg name="camera_left"  default="left/image_raw"/>
    <arg name="camera_right" default="right/image_raw"/>

    <!--Number of corners-->
    <arg name="board_width"   default="8"/>
    <arg name="board_height"  default="5"/>
    <arg name="square_length" default="45.0"/><!--mm-->
    <arg name="image_width"  default="1936"/>
    <arg name="image_height" default="1216"/>

    <!--Calibration modes-->
    <!--0 Checker-->
    <!--1 Charuco-->
    <arg name="method" default="1"/>

    <!--Difference Threshold-->
    <arg name="threshold" default="25"/><!--px-->

    <!--Charuco marker length if required-->
    <arg name="marker_length" default="23.0"/><!--mm-->

    <!--Dictionary ID-->
    <arg name="dictionary" default="0"/><!--DICT_4X4_50-->

    <node name="stereo_calibration_node" pkg="stereo_calibration" type="stereo_calibration_node" output="screen">
        <param name ="save_directory"  value="$(arg save_directory)"/>

        <param name ="camera_left"  value="$(arg camera_left)"/>
        <param name ="camera_right" value="$(arg camera_right)"/>

        <param name ="board_width"  value="$(arg board_width)"/>
        <param name ="board_height" value="$(arg board_height)"/>
        <param name ="square_length" value="$(arg square_length)"/>

        <param name ="image_width"  value="$(arg image_width)"/>
        <param name ="image_height" value="$(arg image_height)"/>

        <param name ="method" value="$(arg method)"/>

        <param name ="threshold" value="$(arg threshold)"/>

        <param name ="marker_length" value="$(arg marker_length)"/>

        <param name ="dictionary" value="$(arg dictionary)"/>
    </node>
</launch>
